### PR TITLE
UI: bugfixes: footer alignment and bad server responses

### PR DESCRIPTION
**Pull request by maxlang**
A bug caused the footer to be misaligned on certain pages.

Before:
screen shot 2016-03-30 at 1 14 59 am

After:
screen shot 2016-03-30 at 1 15 05 am

---

This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/5703)

### CODE REVIEW

**Comment by cuongdo** [comment]
I'm not sure we need to slip this in ASAP. Yes, the footer's position in master isn't ideal.

However, this PR still has a bug in the footer positioning. Specifically, with a fresh new cluster, when the "opt in" banner is showing, the footer is hidden just below the bottom edge of the window:

screenshot 2016-03-30 06 01 58

The footer reappears when I close the banner:

screenshot 2016-03-30 06 02 08

@petermattis: any thoughts here?

**Comment by maxlang** [comment]
I'll take another look, but this is a big visual improvement.

**Comment by cuongdo** [comment]
OK, let's get this in the next build, and also get a fix for the case where the banner is visible later.

:lgtm:

---







*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/5703#-:-KE6oSqZr_sv6rcJlg6B:831331105)*

**Comment by maxlang** [comment]
Added a fix for bad responses from external requests.

File: ui/ts/models/cockroachlabs.ts [hunk 1]
```diff
@@ -128,6 +128,7 @@ module Models {
             m.request<VersionList>({
               url: `${COCKROACHLABS_ADDR}/api/clusters/updates?uuid=${this.clusterID}&version=${nodeStatuses[0].build_info.tag}`,
               config: Utils.Http.XHRConfig,
+              background: true,
```
Comment (cuongdo) [inline ui/ts/models/cockroachlabs.ts]: Do you also need this on the other methods like `unregister`? It's still hanging on those.

**Comment by cuongdo** [comment]
*[ui/ts/util/http.ts, line 6 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5703#-KE6xDGrXG-rAujdP_KR:-KE6xDGrXG-rAujdP_KS:-1384977728) ([raw file](https://github.com/cockroachdb/cockroach/blob/02e341bfd8849d18378f4992deb5975d662c8b0c/ui/ts/util/http.ts#L6)):*
This causes a JS console error for me:

```app.js:10 SyntaxError: Unexpected token h(…)

```

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/5703)*

**Comment by cuongdo** [comment]
*[ui/ts/util/http.ts, line 6 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5703#-KE6xDGrXG-rAujdP_KR:-KE6x_hoqKROi8gYPiBi:-1405862404) ([raw file](https://github.com/cockroachdb/cockroach/blob/02e341bfd8849d18378f4992deb5975d662c8b0c/ui/ts/util/http.ts#L6)):*
Actually, never mind... it seems to be doing what's intended:

```
Error attempting to contact Cockroach Labs server. SyntaxError: Unexpected token h(…)
```

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/5703)*

**Comment by maxlang** [comment]
This is expected - it's actually the error we're getting back from the Cockroach Labs server. We have an error parsing it because it's not JSON, but now we just log and continue, whereas before this error was bubbled up.

**Comment by cuongdo** [comment]
:lgtm: for commit #2

tested on a hacked up version of reg server that sleeps for 10 seconds on all HTTP requests

---

Review status: 0 of 5 files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/5703#-:-KE6vu1eNLqusU3PQYQq:396859460)*

**Pull request merged**